homeassistant/components/velbus.py

```diff
@@ -26,18 +28,26 @@
 }, extra=vol.ALLOW_EXTRA)
 
 
-def setup(hass, config):
+@asyncio.coroutine
+def async_setup(hass, config):
     """Set up the Velbus platform."""
     import velbus
     port = config[DOMAIN].get(CONF_PORT)
-    connection = velbus.VelbusUSBConnection(port)
-    controller = velbus.Controller(connection)
+    controller = velbus.Controller(port)
     hass.data[DOMAIN] = controller
 
     def stop_velbus(event):
         """Disconnect from serial port."""
         _LOGGER.debug("Shutting down ")
-        connection.stop()
+        controller.stop()
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, stop_velbus)
+
+    def callback():
+        """Callback when scan is completed."""
+        hass.async_add_job(async_load_platform(hass, 'switch', DOMAIN))
```

Comment:
Add an empty dict as discovery_info (fourth argument) and the original config as fifth argument.

Reply:
ok. See also other comment: if discovery_info is empty, why should I return on discovery_info being None in the platforms?